### PR TITLE
vdk-server: Wrap installation/uninstallation to catch any uncaught errors

### PR DESCRIPTION
**Pull request by gageorgiev**
This PR wraps both the `installer.install` and `installer.uninstall` in a wrapper function to ensure any uncaught exceptions can still alert the user.

This is a somewhat hacky solution and I'm not 100% clear on why this behavior appears here.

Testing done: raised exception inside installer code manually and checked output

### CODE REVIEW

File: projects/vdk-plugins/vdk-server/src/vdk/plugin/server/server_plugin.py [hunk 1]
```diff
@@ -71,9 +84,9 @@ def server(install, uninstall, status):
     else:
```
Comment (murphp15) [inline projects/vdk-plugins/vdk-server/src/vdk/plugin/server/server_plugin.py:84]: Can you wrap all the code in one block?

at the moment errors in check status will still be silent.

**Review by murphp15 — APPROVED** [review]
Looks good, just made one comment!

File: projects/vdk-plugins/vdk-server/src/vdk/plugin/server/server_plugin.py [hunk 2]
```diff
@@ -56,26 +57,29 @@
...
+            raise Exception("paul test")
+            installer = Installer()
+            if install:
+                installer.install()
+            elif uninstall:
+                installer.uninstall()
+            else:
+                installer.check_status()
+    except Exception as e:
+        log.exception("VDK CLI command failed")
```
Comment (yonitoo) [inline projects/vdk-plugins/vdk-server/src/vdk/plugin/server/server_plugin.py:82]: Why don't we print the caught exception as well

File: projects/vdk-plugins/vdk-server/src/vdk/plugin/server/server_plugin.py [hunk 3]
```diff
@@ -56,26 +57,29 @@
...
+            flags += 1
+        if status:
+            flags += 1
+        if flags != 1:
+            click.echo(
+                "Exactly one of --install, --uninstall, or --status options should be specified"
+            )
         else:
-            installer.check_status()
+            raise Exception("paul test")
```
Comment (yonitoo) [inline projects/vdk-plugins/vdk-server/src/vdk/plugin/server/server_plugin.py:73]: nit: change "paul test"

**Review by yonitoo — COMMENTED** [review]
LGTM overall. Added a few comments.

File: projects/vdk-plugins/vdk-server/src/vdk/plugin/server/server_plugin.py (hunk 2 above)
Comment (murphp15) [reply projects/vdk-plugins/vdk-server/src/vdk/plugin/server/server_plugin.py:82]: The exception is printed when you all this function. Some magic stuff.

File: projects/vdk-plugins/vdk-server/src/vdk/plugin/server/server_plugin.py [hunk 4]
```diff
@@ -4,6 +4,7 @@
 VDK Server plugin script.
 """
 import logging
+from typing import Callable
```
[the remark this answers is not in the document]
Comment (murphp15) [reply projects/vdk-plugins/vdk-server/src/vdk/plugin/server/server_plugin.py]: removed.

File: projects/vdk-plugins/vdk-server/src/vdk/plugin/server/server_plugin.py (hunk 3 above)
Comment (murphp15) [reply projects/vdk-plugins/vdk-server/src/vdk/plugin/server/server_plugin.py]: removed

**Review by yonitoo — APPROVED** [review]
lgtm now

**Pull request merged**